server/container/guice/mailbox/src/main/java/org/apache/james/modules/mailbox/ReIndexerModules.java

```diff
@@ -0,0 +1,39 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.modules.mailbox;
+
+import org.apache.james.mailbox.indexer.MessageIdReIndexer;
+import org.apache.james.mailbox.indexer.ReIndexer;
+import org.apache.mailbox.tools.indexer.MessageIdReIndexerImpl;
+import org.apache.mailbox.tools.indexer.ReIndexerImpl;
+
+import com.google.inject.Module;
+import com.google.inject.Scopes;
+
+public interface ReIndexerModules {
+    Module REINDEXER_MODULE = binder -> {
```

Comment:
could we pick a better name? ReIndexer.REINDEXER doesn't mean anything to me.
What does it reindex?